core/src/main/scala_spark_others/org/apache/spark/h2o/SparkSpecificUtils.scala

```diff
@@ -32,7 +33,24 @@ object SparkSpecificUtils extends CrossSparkUtils {
       content: => Seq[Node],
       activeTab: SparkUITab,
       helpText: String): Seq[Node] = {
-    UIUtils.headerSparkPage(request, "Sparkling Water", content, activeTab, helpText = Some(helpText))
+    val method = UIUtils.getClass.getMethods.find(m => m.getName == "headerSparkPage").get
```

Comment:
This one is obviously fragile and can break if an overloaded method is added to spark.ui

But I guess that you can live with this?

Reply:
Otherwise LGTM

Reply:
Yes, if the method doesn't exist the Spark UI shows an exception if Sparkling Water tab is clicked. For Apache Spark we know that there is such a method. This could if a user deploys SW to a obscure platform with a custom build of Spark which we haven't tested. IMHO, we can live with that.